client/src/components/SignUp.js

```diff
@@ -1,13 +1,177 @@
+      isErrorPassword: false,
+    });
+    const name = this.state;
+    const characters = [
+      '+',
+      '/',
+      '*',
+      '$',
+      '#',
+      '@',
+      '!',
+      '^',
+      '&',
+      '(',
+      ')',
+      '?',
+      '>',
+      '<',
+      '.',
+    ];
+    const errors = {
+      nameError: '',
+      emailError: '',
+      passwordError: '',
+      isErrorName: false,
+      isErrorEmail: false,
+      isErrorPassword: false,
+    };
+    for (let i = 0; i < characters.length; i++) {
+      if (name.includes(characters[i])) {
+        isError = true;
+        errors.isErrorName = true;
+        errors.nameError =
+          'Name should only contain letters, numbers, underscores and dashes.';
+      } else if (name < 1) {
+        isError = true;
+        errors.isErrorName = true;
+        errors.nameError = 'Name is required.';
+      }
+    }
+
+    this.setState({
+      ...this.state,
+      ...errors,
+    });
+    return isError;
+  };
+
+  signup = () => {
+    const err = this.validate();
+    if (!err) {
+      const { history } = this.props;
+      const { name, email, password } = this.state;
+      const inputs = {
+        name,
+        email,
+        password,
+      };
+      axios.post('/account/sign-up', inputs).then(({ data }) => {
+        if (data.success) {
+          history.push({
+            pathname: '/account/login',
+            data: this.props.location.data,
+          });
+        } else {
+          this.setState({
+            nameError: 'This name already exists.',
+            isErrorName: true,
+          });
+        }
+      });
+    }
+  };
 
-class SignUp extends Component {
   render() {
+    const {
+      isErrorEmail,
+      isErrorName,
+      isErrorPassword,
+      nameError,
+      name,
+      emailError,
+      email,
+      passwordError,
+      password,
+    } = this.state;
     return (
       <Fragment>
-        <h1>SignUp</h1>
+        <S.Main>
+          <S.H1>Sign up</S.H1>
+          <S.Form>
+            <S.Label for="name"> Name: </S.Label>
+            <S.Input
+              StyleError={isErrorName}
+              {...this.props}
+              type="text"
+              name="name"
+              label="name"
+              errorText={nameError}
+              value={name}
+              onChange={e =>
+                this.setState({
+                  name: e.target.value,
+                })
+              }
+            />
+            <S.Label for="email"> Email: </S.Label>
+            <S.Input
+              StyleError={isErrorEmail}
+              {...this.props}
+              type="text"
+              name="email"
+              label="email"
+              errorText={emailError}
+              value={email}
+              onChange={e =>
+                this.setState({
+                  email: e.target.value,
+                })
+              }
+            />
+            <S.Label for="password"> Password: </S.Label>
```

Comment:
need to create another input for a second password input to check that someone has entered their password in correctly!